matching-web-bundle/src/main/java/com/graphhopper/matching/http/MapMatchingResource.java

```diff
@@ -99,13 +99,20 @@ public Response match(
 
         PMap hints = createHintsMap(uriInfo.getQueryParameters());
         // add values that are not in hints because they were explicitly listed in query params
-        hints.putObject("vehicle", vehicleStr);
         hints.putObject(MAX_VISITED_NODES, maxVisitedNodes);
-        // resolve profile and remove legacy vehicle/weighting parameters
-        String profile = profileResolver.resolveProfile(hints).getName();
-        hints.remove("vehicle");
-        hints.remove("weighting");
+        String weightingVehicleLogStr = "weighting: " + hints.getString("weighting", "") + ", vehicle: " + hints.getString("vehicle", "");
+        if (Helper.isEmpty(profile)) {
+            // resolve profile and remove legacy vehicle/weighting parameters, we need to explicitly disable CH
+            // here because map matching does not use it
+            boolean hadCHDisable = hints.has(Parameters.CH.DISABLE);
+            hints.putObject(Parameters.CH.DISABLE, true);
+            profile = profileResolver.resolveProfile(hints).getName();
+            if (hadCHDisable)
+                hints.remove(Parameters.CH.DISABLE);
```

Comment:
Couldn't we put this into removeLegacyParameters unconditionally?

Reply:
I think right now we allow `ch.disable=true` to disable LM in `MapMatching`. It basically means "please use Dijkstra because my GPX data is dense and Dijkstra is faster".  So if we remove it we need to add `lm.disable` instead. But now that you say this shouldn't it be `!hadCHDisable` here (we only want to temporarily add ch.disable for `resolveProfile`). Maybe it would be cleaner to copy `hints` and add `ch.disable=true` without modifying `hints`.